packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -476,6 +483,67 @@ - (FlutterError *_Nullable)onListenWithArguments:(id _Nullable)arguments
   return nil;
 }
 
+
+- (NSArray<FLTGetEmbeddedSubtitlesMessage *> *) getEmbeddedSubtitles {
+    NSArray<AVMediaCharacteristic> *characteristics= _player.currentItem.asset.availableMediaCharacteristicsWithMediaSelectionOptions;
+    
+    
+    for(int i=0;i<[characteristics count];i++){
+        AVMediaCharacteristic characteristic = characteristics[i];
+        if([characteristic.description isEqual: @"AVMediaCharacteristicLegible"]){
+            
+            
+            AVMediaSelectionGroup *group = [
+                _player.currentItem.asset mediaSelectionGroupForMediaCharacteristic:characteristic
+            ];
+            NSArray<AVMediaSelectionOption *> *options = [group options];
+            
+            NSMutableArray<FLTGetEmbeddedSubtitlesMessage *> *subtitles = [[NSMutableArray alloc]initWithCapacity:[options count]];
+            
+            for(int j=0;j<[options count];j++){
+                AVMediaSelectionOption *option = options[j];
+                
+                FLTGetEmbeddedSubtitlesMessage *subtitle =
+                    [FLTGetEmbeddedSubtitlesMessage makeWithLanguage:option.locale.localeIdentifier
+                                                               label:option.displayName
+                                                          trackIndex:@(j)
+                                                          groupIndex:@(i)
+                                                         renderIndex:@(2)];
+                
+                [subtitles addObject: subtitle];
+            }
+            
+            return subtitles;
+        }
+    }
+    
+    NSMutableArray<FLTGetEmbeddedSubtitlesMessage *> *subtitles = [[NSMutableArray alloc]initWithCapacity:0];
+    return subtitles;
```

Comment:
can directly return here. Also why are you returning an empty mutable array here?